src/snakemake/pathvars.py

```
@@ -0,0 +1,110 @@
+
+    def get(self, name: str) -> str:
+        return self.items[name]
+
+    def apply(self, path: str) -> str:
+        seen = set()
+        applied_path = path
+        while PATHVAR_REGEX.search(applied_path):
+            if applied_path in seen:
+                raise WorkflowError(
+                    f"Cyclic pathvar reference detected when expanding pathvars in {path}. Last expansion: {applied_path}."
+                )
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)